onnxruntime/core/providers/qnn/builder/opbuilder/conv_op_builder.cc

```diff
@@ -78,154 +107,392 @@ Status ConvOpBuilder::IsOpSupported(QnnModelWrapper& qnn_model_wrapper,
+                                                           is_graph_input));
+      if (conv_type == OnnxConvType::kConv) {
+        ORT_RETURN_IF_ERROR(qnn_model_wrapper.AddNchwToHwcnTranspose(node_unit.Index(),
+                                                                     reshape_output,
+                                                                     conv_weight_input_name,
+                                                                     shape_2d,
+                                                                     final_shape,
+                                                                     input_info.qnn_data_type,
+                                                                     input_info.quant_param,
+                                                                     do_op_validation,
+                                                                     false));
+      } else if (conv_type == OnnxConvType::kConvTranspose) {
+        ORT_RETURN_IF_ERROR(qnn_model_wrapper.AddCnhwToHwcnTranspose(node_unit.Index(),
+                                                                     reshape_output,
+                                                                     conv_weight_input_name,
+                                                                     shape_2d,
+                                                                     final_shape,
+                                                                     input_info.qnn_data_type,
+                                                                     input_info.quant_param,
+                                                                     do_op_validation,
+                                                                     false));
+      } else {
+        return ORT_MAKE_STATUS(ONNXRUNTIME, FAIL, "QNN EP: Unexpected convolution op type: ", node_unit.OpType().c_str());
+      }
+    }
 
-    QnnTensorWrapper input_tensorwrapper(input_tensor_name, tensor_type, qnn_data_type, quantize_param,
-                                         std::move(input_shape), std::move(unpacked_tensor));
+    Qnn_TensorType_t tensor_type = GetInputTensorType(qnn_model_wrapper, conv_weight_input_name);
+    QnnTensorWrapper input_tensorwrapper(conv_weight_input_name, tensor_type, input_info.qnn_data_type,
+                                         input_info.quant_param, std::move(final_shape), std::move(unpacked_tensor));
     ORT_RETURN_IF_NOT(qnn_model_wrapper.AddTensorWrapper(std::move(input_tensorwrapper)), "Failed to add tensor.");
   }
 
-  if (inputs.size() < 3) {  // Add default bais
-    ORT_RETURN_IF_ERROR(AddDefaultBias(qnn_model_wrapper, weight_m, GetNodeName(node_unit),
-                                       is_quantized_model, logger, input_names));
+  //
+  // Input 2: bias
+  //
+  if (num_inputs == 3) {
+    ORT_RETURN_IF_ERROR(ProcessInput(qnn_model_wrapper, inputs[2], logger, is_quantized_model, input_names));
+  }
```

Comment:
The reason we always provide the default bias is QNN run into error if Conv weight is not static. Do you mean it's working now?

Reply:
Yeah, I added unit tests for Conv and ConvTranpose (e.g., TestQDQConvU8S32_DynamicWeight_NoBias) that use dynamic weight and no bias. Looks to work for sdk 2.10. Do you recall which version of QNN SDK had that issue?

Reply:
Or maybe a specific example that produced that error?